### DISCUSSION TITLE
Using Hydrogen GraphQL product search parameters on internationalised shop

### DISCUSSION THREAD

**Discussion by selvinkuik** [Shopify/storefront-api-feedback#171] [Help]
Take the following GQL...

```
query Search(
  $country: CountryCode
  $cursor: String
  $language: LanguageCode
  $pageBy: Int!
  $query: String
) @inContext(country: $country, language: $language) {
  products(after: $cursor, first: $pageBy, query: $query) {
    ...ProductsFragment
  }
}
```

Let's say I have a store in FR and EN. And I have some products which are rings, which would be listed as "bagues" in FR.

Now if I run the above query with...

- country: FR
- language: FR
- query: rings

I am returned results, although my search query was in EN.

And if I run the same query with...

- country: FR
- language: FR
- query: bagues

I get no results. Shopify should respect the language I am searching in and return results based on that language

**Comment by fahjan** [comment]
any updates here?
@selvinkuik

**Comment by selvinkuik** [comment]
@fahjan I ended up building a bespoke client side search in the end

**Reply by fahjan** [reply]
can you provide a hint ? please
@selvinkuik
